File: Bridge/System/Runtime/InteropServices/Marshal.cs
using Bridge;

namespace System.Runtime.InteropServices
{
    [External]
    public static class Marshal
    {
        public static IntPtr AllocHGlobal(IntPtr cb)
        {
            throw new NotImplementedException();
        }

        public static IntPtr AllocHGlobal(int cb)
        {
            return new IntPtr(cb);
        }

        public static void FreeHGlobal(IntPtr hglobal)
        {
            //delete IntPtr, free buffer
            throw new NotImplementedException();
        }

        public static string PtrToStringAnsi(IntPtr ptr)
        {
            throw new NotImplementedException();
        }

        public static string PtrToStringAnsi(IntPtr ptr, int len)
        {
            throw new NotImplementedException();
        }

        public static void PtrToStructure(IntPtr ptr, object structure)
        {
            throw new NotImplementedException();
        }

        public static object PtrToStructure(IntPtr ptr, Type structureType)
        {
            throw new NotImplementedException();
        }

        public static int SizeOf(object structure)
        {
            throw new NotImplementedException();
        }

        public static int SizeOf(Type t)
        {
            throw new NotImplementedException();
        }

        public static void Copy(byte[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(char[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(short[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(int[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(long[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(float[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(double[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr[] source, int startIndex, IntPtr destination, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, byte[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, char[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, short[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, int[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, long[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, float[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, double[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }

        public static void Copy(IntPtr source, IntPtr[] destination, int startIndex, int length)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/System/Runtime/InteropServices/Marshal.js
Bridge.define("System.Runtime.InteropServices.Marshal", {
    statics: {
        methods: {
            allocHGlobal$1: function (cb) { //from IntPtr
                throw new System.NotImplementedException();
            },
            allocHGlobal: function (cb) {
                return new System.IntPtr.$ctor1(cb);
            },
            freeHGlobal: function (hglobal) {
                //delete IntPtr, free buffer
                throw new System.NotImplementedException();
            },
            ptrToStringAnsi: function (ptr) {
                throw new System.NotImplementedException();
            },
            ptrToStringAnsi$1: function (ptr, len) {
                throw new System.NotImplementedException();
            },
            ptrToStructure$1: function (ptr, structure) {
                throw new System.NotImplementedException();
            },
            ptrToStructure: function (ptr, structureType) {
                throw new System.NotImplementedException();
            },
            sizeOf: function (structure) {
                throw new System.NotImplementedException();
            },
            sizeOf$1: function (t) {
                throw new System.NotImplementedException();
            },
            copy: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$1: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$3: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$4: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$5: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$15: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$2: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$14: function (source, startIndex, destination, length) {
                throw new System.NotImplementedException();
            },
            copy$6: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$7: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$9: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$10: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$11: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$13: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$8: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            },
            copy$12: function (source, destination, startIndex, length) {
                throw new System.NotImplementedException();
            }
        }
    }
});
